api-server/src/server/boot/user.js

```diff
@@ -468,6 +474,42 @@ function postResetProgress(req, res, next) {
   );
 }
 
+async function postResetProjectProgress(req, res, next) {
+  try {
+    const { user } = req;
+    const { dashedBlockName } = req.params;
+
+    const toResetChallengeIds = getChallengeIdsForBlock(dashedBlockName);
```

Comment:
This looks like a slow operation, so should run once on boot, rather than on every request.

Reply:
To address the performance concerns raised, I've implemented caching through lazy loading for the challenge block lookup. The `challengeMap` is now generated on-demand by `getChallengeMapByBlock` and stored for subsequent use. This change minimizes redundant computations and optimizes performance by ensuring that the expensive operation of building the challenge map is performed only once and reused for future requests. The updates can be found in commit [48c22278c6e5d0c3573b1d3638f542267a97da71].